pep-9999.rst

```diff
@@ -0,0 +1,252 @@
+PEP: 9999
+Title: Inline variance operators for generic type variables: ``+T`` and ``-T``
+Author: Joren Hammudoglu <jhammudoglu at gmail.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra at gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 24-Jul-2021
+Post-History: 24-Jul-2021
+
+
+Abstract
+========
+
+This PEP proposes a consise syntax for specifying the variance of
+the type parameters of `user-defined generic types
+<https://www.python.org/dev/peps/pep-0484/#user-defined-generic-types>`_.
+This allow declaring a co- or contravariant type parameter with ``+T`` or 
+``-T``, where ``T`` is a type variable.
+
+
+Terminology
+===========
+
+This PEP uses some terms to refer to specific concepts related to Python 
+typing, or type systems in general. For the sake of clarity, this PEP 
+considers the following terms and definitions:
+
+type variable
+    An instance of ``typing.TypeVar`` or ``typing.ParamSpec``, as defined
```

Comment:
I don't think it matters that the variadic behaviour of `ParamSpec` is undefined at this moment, just like it is/will be for `TypeVarTuple` from PEP 646. Both inherit from `typing._TypeVarLike`  (although I'm guessing here for `TypeVarTuple`), where the `__covariant__` and `__contravariant__` attributes are set. So I think that `typing._TypeVarLike` is the best place to overload `__pos__` and `__neg__`, so that they return an e.g. `_TypeVarAlias(_TypeVarLike)` instance with an `__origin__: _TypeVarLike` attribute.